src/index/LocatedTriples.cpp

```diff
@@ -57,36 +57,56 @@ NumAddedAndDeleted LocatedTriplesPerBlock::numTriples(size_t blockIndex) const {
   return {countInserts, blockUpdateTriples.size() - countInserts};
 }
 
-// ____________________________________________________________________________
-// Collect the relevant entries of a LocatedTriple into a triple.
-template <size_t numIndexColumns>
-requires(numIndexColumns >= 1 && numIndexColumns <= 3)
+// Return a `std::tie` of the relevant entries of a row, according to
+// `numIndexColumns` and `includeGraphColumn`. For example, if `numIndexColumns`
+// is `2` and `includeGraphColumn` is `true`, the function returns
+// `std::tie(row[0], row[1], row[2])`.
+template <size_t numIndexColumns, bool includeGraphColumn>
+requires(numIndexColumns >= 1 && numIndexColumns <= IdTriple<0>::NumCols)
```

Comment:
Shouldn't the second condition somehow depend on `row` and not on `IdTriple<0>::NumCols`?

Reply:
It should be a hardcoded `3` as it was before, since the graph column is now handled explicitly.